jaeger-core/src/main/java/com/uber/jaeger/samplers/RemoteControlledSampler.java

```diff
@@ -86,36 +88,31 @@ private void updateSampler() {
       return;
     }
 
-    Sampler newSampler;
-    try {
-      newSampler = extractSampler(response);
-    } catch (UnknownSamplingStrategyException e) {
-      metrics.samplerParsingFailure.inc(1);
-      return; // sampler updateGauge without a new sampler
-    }
+    Sampler oldSampler = sampler;
```

Comment:
Could you please elaborate?
My understanding of the [volatile spec](http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.3.1.4) is that this should be thread safe.